src/ansys/dpf/post/simulation.py

```diff
@@ -0,0 +1,524 @@
+        None
+        """
+        self._active_selection = None
+
+    @property
+    def _time_frequencies(self):
+        """Description of the temporal/frequency analysis of the model."""
+        return self._model.metadata.time_freq_support
+
+    @property
+    def _get_time_freq_precision(self):
+        """Computes a precision for times/frequencies requests based on the underlying support."""
+        if self._time_freq_precision is None:
+            available_values = self.time_freq_support.time_frequencies.data
+            diff = np.diff(available_values, prepend=available_values[0] - 1.0)
+            minimum = np.min(diff)
+            self._time_freq_precision = minimum / 20.0
+        return self._time_freq_precision
+
+    @property
+    def time_freq_support(self):
+        """Description of the temporal/frequency analysis of the model."""
+        return self._time_frequencies
+
+    @property
+    def units(self):
+        """Returns the current time/frequency and distance units used."""
+        return self._units
+
+    def __str__(self):
+        """Get the string representation of this class."""
+        txt = (
+            "%s." % re.sub(r"(?<!^)(?=[A-Z])", " ", type(self).__name__)
+            + "\n\n\nData Sources\n------------------------------\n"
+        )
+        ds_str = self._model._data_sources.__str__()
+        txt += ds_str
+        txt += "\n\n"
+        txt += self._model.__str__()
+        return txt
+
+    def _build_components_from_components(self, base_name, category, components):
+        # Create operator internal names based on components
+        out = []
+        if components is None:
+            out = None
+        else:
+            if isinstance(components, int) or isinstance(components, str):
+                components = [components]
+            if not isinstance(components, list):
+                raise ValueError(
+                    "Argument 'components' must be an int, a str, or a list of either."
+                )
+            for comp in components:
+                if not (isinstance(comp, str) or isinstance(comp, int)):
+                    raise ValueError(
+                        "Argument 'components' can only contain integers and/or strings."
+                    )
+                if isinstance(comp, int):
+                    comp = str(comp)
+                if comp not in self._component_to_id.keys():
+                    raise ValueError(
+                        f"Component {comp} is not valid. Please use one of: "
+                        f"{list(self._component_to_id.keys())}."
+                    )
+                out.append(self._component_to_id[comp])
+
+        # Take unique values and build names list
+        if out is not None:
+            out = list(set(out))
+        if out is None and category == ResultCategory.vector:
+            columns = [base_name + comp for comp in self._component_names[:3]]
```

Comment:
why this?

Reply:
This has been refactored to make a clearer distinction between vector and matrix.